### ISSUE TITLE
Smart Start Provisioning Entry Settings Change During Protocol Switch

### ISSUE THREAD

**Issue by The00Dustin** [zwave-js/zwave-js-ui#3808]
### Checklist

- [X]  I am not using Home Assistant. Or: a developer has told me to come here.
- [X] I have checked the troubleshooting section and my problem is not described there.
- [X] I have read the changelog and my problem is not mentioned there.

### Deploy method

Home Assistant Add-on

### Z-Wave JS UI version

9.14.5

### ZwaveJS version

12.12.3

### Describe the bug

Whenever a smart start provisioning entry is changed from `Z-WAVE` to `Z-WAVE LONG RANGE` and then back, the state of the `S2 Unauthenticated` and `S0 Legacy` checkboxes will be unchecked even if they were checked by default and not previously changed by the user.

### To Reproduce

1. Scan the QR for a device that supports LR
2. Note the state of the mentioned checkboxes (ensure at least one is checked)
3. Switch the provisioning entry to LR
4. Switch the provisioning entry back
5. Note the state of the mentioned checkboxes (observe they are no longer checked)

### Expected behavior

Presumably optimal behavior would be to leave the checkbox values alone when switching between protocols, ignoring the inapplicable checkboxes on the backend.  This way, they will consistently contain their "correct" (default or user selected) value regardless of prior circumstances.  Given that, whether inapplicable checkboxes are hidden, disabled, or left in a modifiable state even though they won't have any effect with the selected protocol becomes a design decision.

### Additional context

I originally observed this with both mentioned checkboxes in a previous version where I believe the checkboxes disappeared when switching to LR and reappeared unchecked when switching back.  I more recently observed it on this version with a device that did not have `S0 Unauthenticated` checked by default and believe the checkboxes unchecked vs disappearing when I switched to LR.  If both recollections are accurate, then it is feasible (though presumably unlikely) that changes to this feature could mean that the bug now only applies to `S2 Unauthenticated`.

**Comment by robertsLando** [comment]
@The00Dustin I cannot reproduce this issue. Could you send me a screen recording?

**Comment by The00Dustin** [comment]
I've been meaning to do this, but I keep forgetting when I'm around my devices.  In case it would be helpful information for reproduction, I've probably only done this in the HA addon.  Is there by chance a dead/dummy/sample LR+Security QR code I could use to create an entry for recording?

**Comment by robertsLando** [comment]
Two dummy codes:

900105405135000000000000000000000000000000000000000000100000000000022000000000000000000000080203
900102283135000000000000000000000000000000000000000100100000000000022000000000000000000000080201

You can import them from QR scan by selecting the text input

**Comment by robertsLando** [comment]
Question is: between step 3 and 4 do you also press save or not? Because if you press save that is expexted to work like that, if not I cannot reprodue

**Comment by The00Dustin** [comment]
So once you asked that question, I thought maybe I was triggering a save by activating the entry, but I got a video recorded from one of the test QR codes you provided without activating the entry, and I am not doing anything to trigger a save as far as I can tell.  That having been said, even if I had, I guess I'm questioning that design decision.  If the user didn't intentionally change the non-LR settings from default, why should they be changed as opposed to remaining default but not visible (or not changeable) since LR shouldn't use them regardless?  Anyway, here's the video you requested:

https://github.com/user-attachments/assets/27fa9d63-9a54-4ff6-bb1f-d7de900c8f25

In this recording, I literally tapped the protocol button and then tapped it again.  I took no other action.

**Comment by robertsLando** [comment]
@The00Dustin Yeah that's what I meant. Unfortunately when you save a LR entity the two security classes are not passed as them don't make any sense on a LR device so when you switch back them are false for that reason.

@AlCalzone thoughts on this? Should I always pass you all the security classes even if it's a LR entity?

**Comment by AlCalzone** [comment]
I think it makes sense to have the driver handle the removal of unnecessary classes.
That said, I think this behavior may come from node-zwave-js? Haven't checked yet.

**Comment by robertsLando** [comment]
> That said, I think this behavior may come from node-zwave-js? Haven't checked yet.

From what I can see on my code I pass all values so I think yes it's the driver that is removing them

**Comment by AlCalzone** [comment]
Can you transfer the issue please?
